### PR TITLE
Gh24 - Input configuration switch

### PR DESCRIPTION
**Pull request by kanjoe24**

### CODE REVIEW

**Comment by kanjoe24** [comment]
Also fixed: https://github.com/rdkcentral/ut-core/issues/31

**Comment by Ulrond** [comment]
Change `register_assert_functions();` to `//register_assert_functions(); /* FIXME: Enable when any change is performed to the UT_ASSERT functions in order to support it, since this always fails we want it outside or normal 100% testing. /*`

File: tests/src/ut_test_kvp.c
Comment (kanjoe24) [reply tests/src/ut_test_kvp.c]: I have done changes, but I have a feeling this can be improved/I have missed something. Can you please help me here.

File: tests/src/ut_test_kvp.c
Comment (Ulrond) [inline tests/src/ut_test_kvp.c:300]: your asserts the wrong way round `UT_ASSERT( pInstance != NULL );`, in this case it will be `NULL` so you want to assert ( I think at least ).

Comment (Ulrond) [reply tests/src/ut_test_kvp.c:300]: After swapping the NULL I noticed you could do this though, since the UT_ASSERT is fall through it just reports errors, you can catch it differently.

```c
 pInstance = ut_getPlatformProfile();
  UT_ASSERT(pInstance != NULL);
  if(pInstance == NULL)
  {
      UT_LOG_ERROR("perhaps platform profile was not passed using -p switch");
      return;
  }
```

File: tests/src/ut_test_kvp.c
Comment (Ulrond) [inline tests/src/ut_test_kvp.c:367]: u don't need this anymore right? Since you never need to load the profile as part of Suite 6 do you?

File: tests/src/ut_test_kvp.c
Comment (Ulrond) [inline tests/src/ut_test_kvp.c:420]: do you need this? Does Suite6 use the input file specified by `-p` it shouldn't? I thought it was loading it itself internally.

**Review by Ulrond — CHANGES_REQUESTED** [review]
minor problems

File: tests/src/ut_test_kvp.c
Comment (Ulrond) [reply tests/src/ut_test_kvp.c:300]: You could add some steps in, so it's clear for logging, in or add comments to the code.

UT_LOG_STEP("Check ut_getPlatformProfile() - Positive");

File: tests/src/ut_test_kvp.c
Comment (Ulrond) [inline tests/src/ut_test_kvp.c:305]: // Check for decoding correctly a field from the platformProfile

**Review by Ulrond — APPROVED** [review]
looks good

**Comment by Ulrond** [comment]
need to wait to merge this after #13  is merged, then swap the branch destination to develop

**Pull request merged**
